torch/_inductor/codegen/common.py

```diff
@@ -1761,6 +1765,7 @@ def generate(
         write: bool = True,
         assignment: bool = True,
         dtype: Optional[torch.dtype] = None,
+        shape: Optional[ShapeType] = None,
```

Comment:
Does it have to be `Optional` here, either? Couldn't the default be `()`?

It makes the code simpler if you're not always checking for `None`.

Reply:
`()` means the variable is a scalar. `None` means we don't know.